app/addons/activetasks/tests/activetasks.storesSpec.js

```diff
@@ -0,0 +1,231 @@
+        activeTasksStore.setSelectedRadio('database_compaction');
+        var storeFilteredtable = activeTasksStore.getFilteredTable(activeTasksStore._collection);
+
+        //parse table and check that it only contains objects with type: database_compaction
+        _.each(storeFilteredtable, function (activeTask) {
+          assert.ok(activeTasksStore.passesRadioFilter(activeTask));
+          assert.ok(activeTask.type === activeTasksStore.getSelectedRadio());
+        });
+
+        //check that the other tasks from collection do not have type: database_compaction
+        fakeFilteredTable = _.filter(activeTasksCollection.table, function (task) {
+          return task.type === 'database_compaction';
+        });
+        assert.deepEqual(fakeFilteredTable.sort(sort), storeFilteredtable.sort(sort));
+      });
+
+      it('should filter the table correctly, by radio -- Indexer', function () {
+        activeTasksStore.setSelectedRadio('indexer');
+        var storeFilteredtable = activeTasksStore.getFilteredTable(activeTasksStore._collection);
+
+        //parse table and check that it only contains objects with type: indexer
+        _.each(storeFilteredtable, function (activeTask) {
+          assert.ok(activeTasksStore.passesRadioFilter(activeTask));
+          assert.ok(activeTask.type === activeTasksStore.getSelectedRadio());
+        });
+
+        //check that the other tasks from collection do not have type: indexer
+        fakeFilteredTable = _.filter(activeTasksCollection.table, function (task) {
+          return task.type === 'indexer';
+        });
+        assert.deepEqual(fakeFilteredTable.sort(sort), storeFilteredtable.sort(sort));
+      });
+
+      it('should filter the table correctly, by radio -- View Compaction', function () {
+        activeTasksStore.setSelectedRadio('view_compaction');
+        var storeFilteredtable = activeTasksStore.getFilteredTable(activeTasksStore._collection);
+
+        //parse table and check that it only contains objects with type: view_compaction
+        _.each(storeFilteredtable, function (activeTask) {
+          assert.ok(activeTasksStore.passesRadioFilter(activeTask));
+          assert.ok(activeTask.type === activeTasksStore.getSelectedRadio());
+        });
+
+        //check that the other tasks from collection do not have type: view_compaction
+        fakeFilteredTable = _.filter(activeTasksCollection.table, function (task) {
+          return task.type === 'view_compaction';
+        });
+        assert.deepEqual(fakeFilteredTable.sort(sort), storeFilteredtable.sort(sort));
+      });
+
+      it('should search the table correctly', function () {
+        activeTasksStore.setSelectedRadio('all_tasks');
+        var searchTerm = 'base';
+        activeTasksStore.setSearchTerm(searchTerm);
+        var storeGeneratedTable = activeTasksStore.getFilteredTable(activeTasksStore._collection);
```

Comment:
Instead of generating a take filterered table. Rather just create a table with the required results. Then check that the store can generate the same results. It will greatly simplify the test